### DISCUSSION TITLE
Any compatible headless CMS?

### DISCUSSION THREAD

**Discussion by linus-ha** [planetscale/discussion#274] [Q&A] [answered]
I am now at a stage where I can decide on what headless CMS to use. 
Possible candidates that unfortunately do not work because of foreign key constrains (vitess requirement):
- Directus
- Strapi
- Ghost

Anyone got something usable working with planet scale? GraphQL/REST is a must. In addition, a customizable WYSIWYG editor and easy asset management would be great. If nothing like this is available (that is compatible) I have to write my own.

EDIT: found "**keystoneJS / Keystone 6"** https://github.com/planetscale/discussion/discussions/274#discussioncomment-4344347

https://keystonejs.com/
https://github.com/keystonejs/keystone

**Comment by fooforge** [comment]
@linus-ha I'm not aware of any, I'm afraid. Most headless CMS are either proprietary or use other data backends (e.g. netlify-cms using Git, Cockpit using Mongo, etc.).

If you decide to roll your own, feel free to drop a link here. I'd love to follow along.

**Reply by linus-ha** [reply]
https://github.com/planetscale/discussion/discussions/274#discussioncomment-4344347

**Comment by amosbastian** [comment]
What did you end up going with?

**Reply by linus-ha** [reply]
https://github.com/planetscale/discussion/discussions/274#discussioncomment-4344347

**Comment by linus-ha** [comment, marked as answer]
https://keystonejs.com/
Automatically generates prisma schema. The underlying prisma client can be configured through keystone to support PlanetScale as described on PlanetScale's website.

Head of my `keystone.ts` file
```js
import { config } from '@keystone-6/core';

import { lists } from './schema';
import { withAuth, session } from './auth';

export default withAuth(
  config({
    db: {
      provider: 'mysql',
      url: process.env.DATABASE_URL ?? "",
      prismaPreviewFeatures: ["referentialIntegrity"],
      additionalPrismaDatasourceProperties: {
        referentialIntegrity: "prisma",
      },
      enableLogging: true,
      useMigrations: false,
      idField: { kind: 'autoincrement' },
    },
[...]
```

**Reply by fooforge** [reply]
Thanks for sharing, @linus-ha!

**Reply by amosbastian** [reply]
Unfortunately they don't support internationalisation so they don't work for my use case 😞 Will keep looking.

**Reply by linus-ha** [reply]
Yeah, right, that's still a big problem. You can work around this problem, by adding a virtual field that takes your desired language as an argument and then looks up the version of the field you wanted. If the requested field is empty (has no translation jet) it could easily fetch a fallback language instead. This can be realized through a function generating the fields (fieldname_languge + virtual field that takes argument), as shown [here](https://github.com/keystonejs/keystone/discussions/7323#discussioncomment-2285289). I hope they work on this soon.

For the meantime, I am implementing this set of fields as described above. Can share this function when I am done testing + being happy with my implementation.
